### PR TITLE
mimic: rbd: use the ordered throttle for the export action

### PR DESCRIPTION
**Pull request by smithfarm**
backport tracker: https://tracker.ceph.com/issues/40500

---

backport of https://github.com/ceph/ceph/pull/28657
parent tracker: https://tracker.ceph.com/issues/40435

this backport was staged using https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by smithfarm** [comment]
All tests passed, and then suddenly for no apparent reason:

```
156/157 Test #128: unittest_alloc_bench ....................   Passed  766.66 sec
./run-make-check.sh: line 36: 24585 Terminated              $DRY_RUN ctest $CHECK_MAKEOPTS --output-on-failure
```

**Comment by smithfarm** [comment]
jenkins test make check

**Comment by yuriw** [comment]
[wip-yuri4-testing-2019-10-04-0016-mimic](https://trello.com/c/sWFPFDBT/709-wip-yuri4-testing-2019-10-04-0016-mimic)

File: src/tools/rbd/action/Export.cc [hunk 1]
```diff
@@ -550,21 +551,20 @@ static int do_export_v1(librbd::Image& image, librbd::image_info_t &info, int fd
...
+                     int export_format)
 {
   librbd::image_info_t info;
   int64_t r = image.stat(info, sizeof(info));
   if (r < 0)
     return r;
 
   int fd;
-  int max_concurrent_ops;
+  int max_concurrent_ops = g_conf->get_val<int64_t>("rbd_concurrent_management_ops");
```
Comment (dillaman) [inline src/tools/rbd/action/Export.cc]: Tweaked this line to fix backport conflict re: `uint64_t` vs `int64_t` type

**Comment by smithfarm** [comment]
@dillaman The compiler now emits a "wall of text" starting with:

```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/tools/rbd/action/Export.cc:1:1: error: 'src' does not name a type
 src/tools/rbd/action/Export.cc
 ^~~
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/tools/rbd/action/Export.cc:5:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/include/compat.h:170:34: error: 'off_t' has not been declared
 int ceph_posix_fallocate(int fd, off_t offset, off_t len);
                                  ^~~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/include/compat.h:170:48: error: 'off_t' has not been declared
 int ceph_posix_fallocate(int fd, off_t offset, off_t len);
                                                ^~~~~
```

File: src/tools/rbd/action/Export.cc [hunk 2]
```diff
@@ -1,3 +1,4 @@
+src/tools/rbd/action/Export.cc
```
Comment (smithfarm) [inline src/tools/rbd/action/Export.cc]: oops ^^^

Comment (dillaman) [reply src/tools/rbd/action/Export.cc]: oops indeed

**Comment by smithfarm** [comment]
@dillaman Thanks for taking care :-)

**Pull request merged**